sympy/functions/special/tests/test_tensor_functions.py

```diff
@@ -57,6 +60,8 @@ def test_kronecker_delta():
     assert adjoint(KroneckerDelta(i, j)) == KroneckerDelta(i, j)
     assert conjugate(KroneckerDelta(i, j)) == KroneckerDelta(i, j)
     assert transpose(KroneckerDelta(i, j)) == KroneckerDelta(i, j)
+    assert KroneckerDelta(n, 0)*cos(n) == KroneckerDelta(n, 0)
```

Comment:
The reason this is not failing is that you're using `n` from the above loop. 
So this test formulation would not be correct. You should define a symbol 'n' before